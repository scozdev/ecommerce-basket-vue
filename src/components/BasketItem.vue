<template>
  <div class="basketItem">
    <section class="left">
      <img :src="cart.image" :alt="cart.name" />
      <div class="count">
        <button>-</button>
        <input type="number" />
        <button>4</button>
      </div>
    </section>

    <div class="right">
      <div>
        <p>{{ cart.name }}</p>
        <strong>{{ cart.price }}</strong>
      </div>
      <button @click="removeItemFromCart(cart.id)">REMOVE</button>
    </div>
  </div>
</template>

<script>
import { mapActions } from "vuex";
export default {
  name: "BasketItem",
  components: {},
  props: ["cartData"],
  data() {
    return {
      cart: this.cartData,
    };
  },
  methods: {
    ...mapActions(["removeItemFromCart"]),
  },
};
</script>

<style scoped>
.basketItem {
  border: 1px solid rgba(0, 0, 0, 0.2);
  text-align: center;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
img {
  width: 100%;
}

.left {
  width: 100px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 5px;
}
.left > * {
  padding: 5px;
}
.right {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.count {
  display: flex;
}
.count input {
  width: 40px;
}
</style>
