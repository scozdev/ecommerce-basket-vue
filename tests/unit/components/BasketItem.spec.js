import { mount, RouterLinkStub, shallowMount } from '@vue/test-utils';
import BasketItem from '@/components/BasketItem';

describe('BasketItem.vue', () => {
  let product;
  let wrapper;

  beforeEach(() => {
    product = {
      id: 1,
      name: 'Åžampuan',
      price: '13',
      currency: 'TRY',
      image:
        'https://cdn.glitch.com/a28552e7-44e1-4bbd-b298-5745e70c2209%2Fsampuan.jpeg?v=1561027551321',
    };

    wrapper = shallowMount(BasketItem, {
      propsData: {
        cartData: product,
      },
    });
  });

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('class property test', () => {
    expect(wrapper.classes()).toContain('basketItem');
  });
});
