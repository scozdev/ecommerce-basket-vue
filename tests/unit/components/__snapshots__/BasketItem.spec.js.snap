// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketItem.vue should match snapshot 1`] = `
<div class="basketItem">
  <section class="left"><img src="https://cdn.glitch.com/a28552e7-44e1-4bbd-b298-5745e70c2209%2Fsampuan.jpeg?v=1561027551321" alt="Şampuan">
    <div class="count"><button>-</button><input type="number"><button>+</button></div>
  </section>
  <div class="right">
    <div>
      <p>Şampuan</p><strong>13</strong>
    </div><button>REMOVE</button>
  </div>
</div>
`;
